src/foam/dao/AbstractDAO.js

```diff
@@ -284,6 +284,12 @@ foam.CLASS({
   ],
 
   methods: [
+    function find(key) {
```

Comment:
Would it make more sense to pass predicate as an extra parameter to find() the same way we do on .select()?

Slight advantage here is if you're going over the network we don't pay to send objects that you're going to filter out on the client.  Less important for find() since you're getting only a single object.